demo/Dockerfile

```diff
@@ -1,29 +1,40 @@
-FROM dockerfile/java
-RUN curl -o /tmp/maven.tgz http://mirror.cogentco.com/pub/apache/maven/maven-3/3.2.2/binaries/apache-maven-3.2.2-bin.tar.gz
-RUN cd /usr/local; tar xvzf /tmp/maven.tgz
-RUN mkdir -p /var/lib/jenkins/plugins /var/lib/jenkins/jobs/cd
+FROM ubuntu:utopic
+RUN apt-get update
+RUN apt-get install -y openjdk-8-jre-headless
+# TODO workaround for nonexistence of libcurl3-gnutls_7.37.1-1ubuntu3.1_amd64.deb mentioned in first update
+RUN apt-get update
+RUN apt-get install -y --fix-missing openssh-server wget git
+RUN echo 'root:root' | chpasswd
+
+# installed maven package is still 3.0.5, pretty old
+ENV MAVEN_VERSION 3.3.1
+RUN cd /usr/local; wget -O - http://mirrors.ibiblio.org/apache/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz | tar xvzf -
 
 # install Jetty
 WORKDIR /opt
-ENV VERSION 9.2.2.v20140723
-RUN curl "http://mirrors.ibiblio.org/eclipse/jetty/$VERSION/dist/jetty-distribution-$VERSION.tar.gz" | tar xvfz -
-RUN ln -sv jetty-distribution-$VERSION jetty
+# jetty package is still 8
+ENV JETTY_VERSION 9.2.9.v20150224
+RUN wget -O - "http://mirrors.ibiblio.org/eclipse/jetty/$JETTY_VERSION/dist/jetty-distribution-$JETTY_VERSION.tar.gz" | tar xvfz -
+RUN ln -sv jetty-distribution-$JETTY_VERSION jetty
 RUN cd /tmp; ln -s /opt/jetty/webapps
 
-# install OpenSSH and allow root to login with password 'root'
-RUN apt-get install -y openssh-server
-RUN echo 'root:root' | chpasswd
+# https://github.com/carlossg/docker-maven/issues/3
+RUN apt-get install -y ca-certificates-java && update-ca-certificates -f
+RUN apt-get install -y openjdk-8-jdk
```

Comment:
Yes a bunch of stuff is in illogical places, to avoid blowing out the Docker cache rerunning everything from the start: some of these installations are quite slow.